daemon/daemon_unix.go

```diff
@@ -550,7 +550,7 @@ func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.
 	// check for various conflicting options with user namespaces
 	if daemon.configStore.RemappedRoot != "" && hostConfig.UsernsMode.IsPrivate() {
 		if hostConfig.Privileged {
-			return warnings, fmt.Errorf("Privileged mode is incompatible with user namespaces")
+			return warnings, fmt.Errorf("Privileged mode is incompatible with user namespaces.  To temporarily disable user namespaces, use '--userns=host'.")
```

Comment:
When we did that change I had the feel that doing in that way we lose a lot of useful debugging information. Perhaps we could introduce some way to enable additional error messages "client" side so that the cli can enrich its own errors and other clients can do that as well. Wdyt ?